### PR TITLE
send aggregated service and census data to event server

### PR DESCRIPTION
**Pull request by mwrock**
This sends data from both the `Service` struct iand the `CensusMember` to the event server. The actual message is now a `ServiceUpdate` instead of a `CensusEntry`. The data is intended to represent a snapshot of key service information. Both the `Service` and `CensusMember` include key data points valuable to an event subscriber. This also exposes the `CensusMember` fields as public. This makes sense since this struct is really intended to expose key census data and consumers will need direct access to this data.

Signed-off-by: Matt Wrock <matt@mattwrock.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by mwrock** [comment]
fixes #2982

File: components/eventsrv-protocol/protocols/event.proto [hunk 1]
```diff
@@ -81,44 +81,43 @@ message CensusEntry {
...
+  optional string topology = 12;
+  /// Update strategy used by the service
+  optional string update_strategy = 13;
+  /// Service group application name
+  optional string application = 14;
+  /// Service group environment name
+  optional string environment = 15;
   /// `true` if this service instance is the leader when in a leader topology
-  optional bool leader = 8;
+  optional bool leader = 16;
```
Comment (raskchanky) [inline components/eventsrv-protocol/protocols/event.proto]: I was told once upon a time that new fields should always be added at the end of a protobuf message because you don't want to change the existing field order.  Is that still the case or did that change?

Comment (christophermaier) [reply components/eventsrv-protocol/protocols/event.proto]: It's not that the physical order in the definition is important, it's the field numbers themselves. If those change then I think you get into situations where you lose your backward-compatibility.

TL;DR: `leader` should stay as field 8.

Comment (mwrock) [reply components/eventsrv-protocol/protocols/event.proto]: Since the very message itself is changing from a CensusEntry to a ServiceUpdate I did not consider compatibility since that alone will break it. @reset and I discussed that a breaking change is acceptable here.

Comment (christophermaier) [reply components/eventsrv-protocol/protocols/event.proto]: Ah, so it is! Yeah, this is fine; sorry!

Comment (raskchanky) [reply components/eventsrv-protocol/protocols/event.proto]: Ah, thanks for clarifying.  Makes sense.

Comment (reset) [reply components/eventsrv-protocol/protocols/event.proto]: Thank you all for being on guard about this! In this case, yeah, it's a do-over so we can break :)

File: components/sup/src/census.rs [hunk 2]
```diff
@@ -468,27 +466,26 @@ impl CensusGroup {
...
+    pub follower: bool,
+    pub update_leader: bool,
+    pub update_follower: bool,
+    pub election_is_running: bool,
+    pub election_is_no_quorum: bool,
+    pub election_is_finished: bool,
+    pub update_election_is_running: bool,
+    pub update_election_is_no_quorum: bool,
+    pub update_election_is_finished: bool,
+    pub alive: bool,
```
Comment (reset) [inline components/sup/src/census.rs]: The health portion here (`alive`, `suspect`, `confirmed`, `departed`) should remain private

**Comment by reset** [comment]
@mwrock one super small nit-pick but this is 100% on point aside from that. Excellent cleanup!

tenor-130581358

File: components/sup/src/census.rs (hunk 2 above)
Comment (mwrock) [reply components/sup/src/census.rs]: ok. I'll need to remove that info from the `ServiceUpdate` message as well unless we think those fields are going to be needed soon by our subscriber. Currently they are not.

Comment (mwrock) [reply components/sup/src/census.rs]: ignore that. I can't read.

**Comment by mwrock** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/269792037).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/269792037).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265
